src/hooks/useFocusEnter.js

```diff
@@ -0,0 +1,16 @@
+import { useCallback, useRef } from 'react'
+
+/* eslint-disable react-hooks/rules-of-hooks */
+export function useFocusEnter(cb, ref = useRef()) {
+  /* eslint-enable react-hooks/rules-of-hooks */
+  const handleFocusEnter = useCallback(
+    event => {
+      if (ref.current.contains(event.target)) {
```

Comment:
We should check for `ref.current` and `ref.current.contains`:
```suggestion
      if (ref.current?.contains?.(event.target)) {
```